Dockerfile

```diff
@@ -1,6 +1,6 @@
 # Build the spiffe-helper binary
-ARG go_version
-FROM --platform=$BUILDPLATFORM golang:${go_version}-alpine as base
+ARG go_version=1.22.3
```

Comment:
why equal is required here?
version must came from https://github.com/spiffe/spiffe-helper/blob/main/Makefile#L90C1-L91C1
or https://github.com/spiffe/spiffe-helper/blob/main/.github/workflows/pr_build.yaml#L79
are we missing something from workflows?

Reply:
Its a default value in case it's not specified. If we don't put it, then we get this warning:

 `- InvalidDefaultArgInFrom: Default value for ARG golang:${go_version}-alpine results in empty or invalid base image name (line 3)`

Reply:
but that means that we have a place where it is not set right?
do you know what call is not adding value?
or it is when you manually run a docker build?
my issue here, is that we are adding another place to update every time we upgrade golang

Reply:
If you run `make docker-build` the warning will appear. The container will still build successfully.

Reply:
Removed the default version.

Reply:
Why was it removed? It sounded like having it was correct.

Reply:
@keeganwitt I opened issue https://github.com/spiffe/spiffe-helper/issues/181 to track the dockerfile warnings. I think its a valid point about it being another thing to keep up to date.